Resources/Controllers/BottomSheet/README.md

```diff
@@ -0,0 +1,159 @@
+#  BottomSheet
+
+A UI component that effectively replaces and extends native UIActionSheet.  
+
+## Overview  
+
+BottomSheet has the following characteristics:
+- dynamically expandable based on the content height
+- swipe down gesture to dismiss
+- tap gesture behind the content to dismiss
+- custom transition animations
+
+### Components:
+
+- **BottomSheet**: *Main component that acts as a "parent" view controller for all Steps*
+```swift
+class BottomSheet<ContentView: BottomSheetContentView>: UIViewController
+```
+- **BottomSheetContentView**: *Main content view*
+```swift
+class BottomSheetContentView: UIView
+```
+- **BottomSheetStep**: *Protocol that defines every Step*
+```swift 
+public protocol BottomSheetStep: UIViewController
+```
+- **BottomSheetDataSource**: *Class that holds a list of all Steps and is handling navigation from Step to Step (next and previous)*
+```swift
+class BottomSheetDataSource: NSObject
+```
+- **LazyStep**: 
+```swift
+typealias LazyStep = () -> BottomSheetStep
+```
+- **AnimatorFactory**: *Enum that holds predefined transition Animations*
+```swift
+enum AnimatorFactory
+```
+
+### Default animations:
```

Comment:
I would change wording from `Default` to `Provided` or `Predefined` or smth similar.